tools/http-fault-injector/Azure.Sdk.Tools.HttpFaultInjector/Program.cs

```diff
@@ -0,0 +1,334 @@
+using CommandLine;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Connections.Features;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Primitives;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Sockets;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace HttpFaultInjector
+{
+    public static class Program
+    {
+        private static HttpClient _httpClient;
+
+        private static readonly List<(string Option, string Description)> _selectionDescriptions = new List<(string Option, string Description)>()
+        {
+            ("f", "Full response"),
+            ("p", "Partial Response (full headers, 50% of body), then wait indefinitely"),
+            ("pc", "Partial Response (full headers, 50% of body), then close (TCP FIN)"),
+            ("pa", "Partial Response (full headers, 50% of body), then abort (TCP RST)"),
+            ("pn", "Partial Response (full headers, 50% of body), then finish normally"),
+            ("n", "No response, then wait indefinitely"),
+            ("nc", "No response, then close (TCP FIN)"),
+            ("na", "No response, then abort (TCP RST)")
+        };
+
+        private static readonly string[] _excludedRequestHeaders = new string[] {
+            // Only applies to request between client and proxy
+            "Proxy-Connection",
+            _responseSelectionHeader
+        };
+
+        // Headers which must be set on HttpContent instead of HttpRequestMessage
+        private static readonly string[] _contentRequestHeaders = new string[] {
+            "Content-Length",
+            "Content-Type",
+        };
+
+        private const string _responseSelectionHeader = "x-ms-faultinjector-response-option";
```

Comment:
We should document that this header can be used for automated scenarios and if it is passed it will be stripped from the request before it is sent to the upstream service.

Reply:
To go along with this could we offer paths for interactive and automated modes?

Reply:
Do you mean just document this in the README.md?  Would you mind submitting another PR to add this after my PR is merged?

I don't know what you mean by "paths for interactive and automated modes"?